internal/weapons/catalyst/moonglow/moonglow.go

```diff
@@ -0,0 +1,65 @@
+package moonglow
+
+import (
+	"fmt"
+
+	"github.com/genshinsim/gcsim/pkg/core"
+)
+
+func init() {
+	core.RegisterWeaponFunc("everlastingmoonglow", weapon)
+	core.RegisterWeaponFunc("everlasting moonglow", weapon)
+	core.RegisterWeaponFunc("moonglow", weapon)
+}
+
+func weapon(char core.Character, c *core.Core, r int, param map[string]int) string {
+	mheal := make([]float64, core.EndStatType)
+	mheal[core.Heal] = 0.075 + float64(r)*0.025
+	char.AddMod(core.CharStatMod{
+		Key: "moonglow-heal-bonus",
+		Amount: func() ([]float64, bool) {
+			return mheal, true
+		},
+		Expiry: -1,
+	})
+
+	matk := make([]float64, core.EndStatType)
+	matk[core.ATK] = (0.0005 + float64(r)*0.0005) * char.MaxHP()
+	char.AddPreDamageMod(core.PreDamageMod{
+		Key: "moonglow-na-bonus",
+		Amount: func(atk *core.AttackEvent, t core.Target) ([]float64, bool) {
+			return matk, (atk.Info.AttackTag == core.AttackTagNormal)
+		},
+		Expiry: -1,
+	})
+
+	icd, dur := -1, -1
+	c.Events.Subscribe(core.PostBurst, func(args ...interface{}) bool {
```

Comment:
just as an FYI, PostBurst is called AFTER the animation frames. i'm not sure how moonglow works but if this is not desirable behavior then might want to hook onto PreBurst

Reply:
> PostBurst is called AFTER the animation frames
oh, i'll replace to PreBurst